fighters/snake/src/status.rs

```diff
@@ -0,0 +1,737 @@
+use super::*;
+use globals::*;
+use smashline::*;
+ 
+
+pub fn install() {
+    install_status_scripts!(
+        // snake_c4_status_pre,
+        // snake_c4_fall_status_main,
+        // snake_down_special_start_status_main,
+        // snake_down_special_start_status_end,
+        // snake_down_special_ground_status_main,
+        // snake_down_special_air_ground_status_main,
+        // snake_down_special_ground_status_end,
+        // snake_down_special_air_ground_status_end,
+        // snake_down_special_status_main,
+        // snake_landing_attack_status_main,
+        snake_side_smash_status_main,
+        snake_side_smash_status_end,
+        snake_side_special_status_main,
+        // snake_side_special_status_end
+        snake_grab_pull_status_main,
+        snake_grab_dash_pull_status_main,
+        snake_grab_attack_status_main,
+        // snake_grab_throw_status_main,
+        snake_grab_wait_status_main,
+        snake_grab_wait_status_end,
+        snake_taunt_status_main,
+        snake_taunt_status_end,
+        snake_down_taunt_wait_status_main,
+        snake_down_taunt_wait_status_end,
+        snake_down_taunt_end_status_main,
+        snake_down_taunt_end_status_end
+    );
+}
+
+//implimented function for checking if an article is "constrained" to snake
+extern "C" {
+    #[link_name = "\u{1}_ZN3app24FighterSpecializer_Snake21is_constraint_articleERNS_7FighterEiNS_22ArticleOperationTargetE"]
+    pub fn is_constraint_article(
+        arg1: *mut smash::app::Fighter,
+        arg2: i32,
+        arg3: smash::app::ArticleOperationTarget,
+    ) -> bool;
+}
+
+#[status_script(agent = "snake_c4", status = WEAPON_SNAKE_C4_STATUS_KIND_ESTABLISH_TARGET, condition = LUA_SCRIPT_STATUS_FUNC_STATUS_PRE)]
+unsafe fn snake_c4_status_pre(weapon: &mut L2CFighterCommon) -> L2CValue {
+    StatusModule::init_settings(
+        weapon.module_accessor,
+        SituationKind(*SITUATION_KIND_AIR),
+        *WEAPON_KINETIC_TYPE_NONE,
+        *GROUND_CORRECT_KIND_AIR as u32,
+        GroundCliffCheckKind(0),
+        false,
+        *WEAPON_STATUS_WORK_KEEP_FLAG_SNAKE_C4_DROP_FALL_FLAG, // <-- wrong flag make it check ground collision?
+        *WEAPON_STATUS_WORK_KEEP_FLAG_SNAKE_C4_ESTABLISH_TARGET_INT,
+        *WEAPON_STATUS_WORK_KEEP_FLAG_SNAKE_C4_ESTABLISH_TARGET_FLOAT,
+        0,
+    );
+    0.into()
+}
+
+
+#[status_script(agent = "snake", status = FIGHTER_SNAKE_STATUS_KIND_SPECIAL_LW_PRODUCE, condition = LUA_SCRIPT_STATUS_FUNC_STATUS_MAIN)]
+unsafe fn snake_down_special_start_status_main(fighter: &mut L2CFighterCommon) -> L2CValue {
+    if WorkModule::is_flag(fighter.module_accessor, *FIGHTER_SNAKE_STATUS_SPECIAL_LW_FLAG_SQUAT) {
```

Comment:
I'd prefer if all of these were `fighter.is_flag` and associated getters/setters, `fighter.module_accessor` gets old.